plugins/inputs/influxdb_v2_listener/influxdb_v2_listener_test.go

```diff
@@ -216,17 +216,21 @@ func TestWriteNoNewline(t *testing.T) {
 	require.NoError(t, listener.Start(acc))
 	defer listener.Stop()
 
-	// post single message to listener
-	resp, err := http.Post(createURL(listener, "http", "/api/v2/write", "bucket=mybucket"), "", bytes.NewBuffer([]byte(testMsgNoNewline)))
-	require.NoError(t, err)
-	require.NoError(t, resp.Body.Close())
-	require.EqualValues(t, 204, resp.StatusCode)
+	for _, parser := range []string{"internal", "upstream"} {
```

Comment:
I've been thinking about this pattern. It reuses the listener for both parsers and always runs in a specific order of parsers. It's best to test as close to real use as possible so this isn't ideal. I think it would be better to pull the initialization into the parser for loop to use a new listener for each parser.

Doing it this way has some testing usability drawbacks too. If there is a failure we won't know which parser was involved. Also it doesn't allow us to run all tests of just one of the parser types. Using golang subtests would fix both those problems. See https://go.dev/blog/subtests#table-driven-tests-using-subtests

Maybe these improvements aren't needed, but they were on my mind so I thought I'd share them with you.

Reply:
Looks good with the Run func.

Could we define the testCases slice in one place, maybe global, so it's not repeated in each function?